### PR TITLE
Upgrade go-systemd and add common target unit

### PR DESCRIPTION
**Pull request by sargun**
### Description of the Change

**FIXME**: please add at least one sentence explaining why changes here are being made.

### CODE REVIEW

**Comment by sargun** [comment]
We should be able to add a drop-in unit in the titus-virtual-kubelet saying that the target is PartOf titus-executor@default__%i.service, and then when that's stopped by VK, everything should "just work"

File: executor/runtime/docker/docker_linux.go [hunk 1]
```diff
@@ -59,6 +60,11 @@ const (
 )
 
 var systemServices = []serviceOpts{
+	{
+		unitName: "titus-container",
```
Comment (solarkennedy) [inline executor/runtime/docker/docker_linux.go:64]: Would this be better named, `titus-sidecar-container` ?

Or, is titus-executor PartOf titus-container?

Why do we need another unit though, why not just have things part of titus-executor?

File: executor/runtime/docker/docker_linux.go [hunk 2]
```diff
@@ -245,7 +270,11 @@ func runServiceInitCommand(ctx context.Context, log *logrus.Entry, cID string, r
 }
 
 func startSystemdUnit(ctx context.Context, conn *dbus.Conn, taskID string, cID string, runtime string, opts serviceOpts) error {
-	qualifiedUnitName := fmt.Sprintf("%s@%s.service", opts.unitName, taskID)
+	postFix := "service"
+	if opts.target {
```
Comment (solarkennedy) [inline executor/runtime/docker/docker_linux.go:274]: Why can't we just put `.service` or `.target` in the unit name itself?

**Review by solarkennedy — APPROVED** [review]

**Comment by solarkennedy** [comment]
Can you confirm: if something fails to stop, is it even bad? (does it bubble up as an error to the user)
Until recently 'titus-sshd@container.service stop' didn't work, there might be others.
Plus, if it is already dead, does it still complain with an error?

Basically I want to make sure this change doesn't the existing behavior of "not caring too much" if sidecars can't be stopped. The show must go on here.

File: executor/runtime/docker/docker_linux.go (hunk 2 above)
Comment (sargun) [reply executor/runtime/docker/docker_linux.go:274]: and parse / unparse it here? That seems uglier than just having a flag for it.

File: executor/runtime/docker/docker_linux.go (hunk 1 above)
Comment (sargun) [reply executor/runtime/docker/docker_linux.go:64]: the target unit isn't a sidecar though. titus-container-sidecars@.target?

File: executor/runtime/docker/docker_linux.go (hunk 2 above)
Comment (solarkennedy) [reply executor/runtime/docker/docker_linux.go:274]: No, just not even parse it at all. Why do we need to parse it?
Just print the whole unit name

Comment (sargun) [reply executor/runtime/docker/docker_linux.go:274]: because you need to have something that inserts the container ID between the "@" and the "."

**Comment by sargun** [comment]
> Can you confirm: if something fails to stop, is it even bad? (does it bubble up as an error to the user)
> Until recently '[titus-sshd@container.service](mailto:titus-sshd@container.service) stop' didn't work, there might be others.
> Plus, if it is already dead, does it still complain with an error?
> 
> Basically I want to make sure this change doesn't the existing behavior of "not caring too much" if sidecars can't be stopped. The show must go on here.

Correct

**Pull request merged**